app/src/main/java/com/hover/stax/transfers/TransferFragment.java

```diff
@@ -206,11 +212,22 @@ protected void onContactSelected(int requestCode, StaxContact contact) {
 		}
 	};
 
-	private void load(Request r) {
+	private void loadAndIndicateFieldState(Request r) {
```

Comment:
This name is much worse. The field state is secondary effect of loading the request. As the Clean Code book puts it, it is a different level of abstraction. If I'm reading this code with no idea what is going on I can understand what load(Request) means but I have no idea what a field state is.

This is a code smell for me that tells me something weird is going on. Why would we set the fieldstate manually for a request, compared to anything else?

Reply:
You're right about it being a different level of abstraction. But I put this here because of the direct effect of loading a request. Based on the design, how I understood and implemented the logic was:
- if a user wants to fulfil a request; Account dropdown, and account number fields should turn green by default without users input, so far as the data are not empty. 

- This is currently the only scenario where a field visibly shows success field state, which was why I set the fieldstate manually

Reply:
Ah, gotcha. That is not correct, the green is for any field that is already valid when the user comes to the form. So when the user has a defaultAccount the channel dropdown will be green whether the user is fulfilling a request or just sending money. There are other fields where this is also true: When creating a new request, if the account number is filled out it should be green. And then I believe that all fields are supposed to turn green once the user has filled them out, which is why triggering in the observer is the way to go.

Reply:
Hmmm, maybe we should ask Justin for his thoughts as well.